internal/processrunner/processrunner.go

```diff
@@ -40,34 +47,55 @@ type processRunner struct {
 type NewProcessRunnerOpts struct {
 	Title   string
 	BinPath string
-	Env     []string
+	EnvPath string
 	Args    []string
 }
 
 // NewProcessRunner creates a new ProcessRunner.
-// It creates a new exec.Cmd with the given binPath and args, and sets the environment.
+// It creates a new exec.Cmd with the given binPath and args, and sets the
+// environment. If no envPath is provided, it uses the current environment using
+// os.Environ().
 func NewProcessRunner(ctx context.Context, opts NewProcessRunnerOpts) ProcessRunner {
+	var env []string
+	if opts.EnvPath != "" {
+		env = GetEnvironment(opts.EnvPath)
+	} else {
+		env = os.Environ()
+	}
 	// using exec.CommandContext to allow for cancellation from caller
 	cmd := exec.CommandContext(ctx, opts.BinPath, opts.Args...)
-	cmd.Env = opts.Env
+	cmd.Env = env
 	return &processRunner{
 		ctx:       ctx,
 		cmd:       cmd,
 		title:     opts.Title,
 		didStart:  make(chan bool),
 		outputBuf: &safebuffer.SafeBuffer{},
 		opts:      opts,
+		env:       env,
 	}
 }
 
+func (pr *processRunner) GetBinPath() string {
```

Comment:
`GetBinPath`, `GetEnvironmentPath`, and `GetEnvironment` aren't used outside of ProcessRunner. I don't think these getters are needed.